src/memory/virtualMemory.ts

```diff
@@ -16,6 +16,9 @@ export class VirtualMemory {
     SegmentAddress
   >();
 
+  private ebp: number;
+  private esp: number;
```

Comment:
By the way, the prefix `e` suggests that we are using a 32-bit x86 architecture but our variables are 8 bytes and not 4.